### ISSUE TITLE
Correct signature for JSONSerializer to_dict and JSONDeserializer from_dict

### ISSUE THREAD

**Issue by calohmn** [benbenbang/types-confluent-kafka#166]
### Initial Checks

- [X] I confirm that I'm using Types-Confluent-Kafka

### Description

This is the same issue as in #154, only it applies to the JSON instead of Avro Serializer and Deserializer.

### Concerning JSONSerializer "to_dict" parameter

Running mypy checks on the following code:
```python
def object_to_dict(obj: object, _ctx: SerializationContext) -> dict[Any, Any]:
    return dict()

JSONSerializer(schema_str, schema_registry_client, to_dict=object_to_dict)
```
produces this error:
> error: Argument "to_dict" to "JSONSerializer" has incompatible type "Callable[[object, SerializationContext], dict[Any, Any]]"; expected "Callable[[object], SerializationContext] | None"  [arg-type]

This is although the docstring in the `confluent_kafka` module `schema_registry/json_schema.py` states that:
> to_dict (callable, optional): Callable(object, SerializationContext) -> dict.

### Concerning JSONDeserializer "from_dict" parameter

Running mypy checks on the following code:
```python
def dict_to_object(obj: dict[Any, Any], _ctx: SerializationContext) -> object:
    return object()

JSONDeserializer(schema_str, from_dict=dict_to_object)
```
produces this error:
> error: Argument "from_dict" to "JSONDeserializer" has incompatible type "Callable[[dict[Any, Any], SerializationContext], object]"; expected "Callable[[dict[Any, Any]], SerializationContext] | None"  [arg-type]

This is although the docstring in the `confluent_kafka` module `schema_registry/json_schema.py` states that:
> from_dict (callable, optional): Callable(dict, SerializationContext) -> object.
---

Comparing the docstrings in the `confluent_kafka` module `schema_registry/json_schema.py` and `schema_registry/avro.py` files, the `from_dict`/`to_dict` types are the same in Avro and JSON Schema classes.

Therefore the Avro fix (#156) needs to be applied also in `confluent_kafka-stubs/schema_registry/json_schema.pyi`.

### Python, verion of Types-confluent-kafka. verion of Confluent-kafka-python, your applications settings & OS Version

```Text
confluent-kafka: 2.3.0
librdkafka: 2.3.0
types-confluent-kafka: 1.2.0
```

### Affected Components

- [X] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`)
- [ ] Apache Kafka broker version
- [ ] Client configuration: `{...}` is provided
- [ ] Operating system is provided
- [ ] Provide IDE screenshot if it's possible

**Comment by benbenbang** [comment]
Thanks @calohmn. Are you ok with opening a PR?

**Comment by benbenbang** [comment]
@calohmn, feel free to leave any comment if there's still something ko.

https://github.com/benbenbang/types-confluent-kafka/pull/169

I'll merge it tomorrow morning CET if there is no news from you 🙁

**Comment by calohmn** [comment]
@benbenbang Thanks! The PR looks good to me.

BTW, when comparing `json_schema.pyi` with `avro.pyi`, I noticed there is a `-> None` return type definition in both the `__call__` methods in `avro.pyi`. This doesn't look correct to me.

**Comment by benbenbang** [comment]
🤦🏻‍♂️  ok, let me take a look

**Comment by benbenbang** [comment]
should be `bytes | None`

**Comment by benbenbang** [comment]
I will close this one and fix the return type in #176
